lib/ansible/modules/network/cloudengine/ce_ospf.py

```diff
@@ -0,0 +1,996 @@
+
+    def get_existing(self):
+        """get existing info"""
+
+        if not self.ospf_info:
+            return
+
+        self.existing["process_id"] = self.process_id
+        self.existing["areas"] = self.ospf_info["areas"]
+        self.existing["nexthops"] = self.ospf_info["nexthops"]
+        self.existing["max_load_balance"] = self.ospf_info.get(
+            "maxLoadBalancing")
+
+    def get_end_state(self):
+        """get end state info"""
+
+        ospf_info = self.get_ospf_dict(self.process_id)
+        if not ospf_info:
+            return
+
+        self.end_state["process_id"] = self.process_id
+        self.end_state["areas"] = ospf_info["areas"]
+        self.end_state["nexthops"] = ospf_info["nexthops"]
+        self.end_state["max_load_balance"] = ospf_info.get("maxLoadBalancing")
+
+    def work(self):
+        """worker"""
+
+        self.check_params()
+        self.ospf_info = self.get_ospf_dict(self.process_id)
+        self.get_existing()
+        self.get_proposed()
+
+        # deal present or absent
+        if self.state == "present":
+            if not self.ospf_info:
+                # create ospf process
+                self.create_process()
+            else:
+                # merge ospf
+                self.merge_process()
+        else:
+            if self.ospf_info:
+                if self.area:
+                    if self.addr:
+                        # remove ospf area network
+                        self.remove_area_network()
+                    else:
+                        # remove ospf area
+                        self.remove_area()
+                if self.nexthop_addr:
+                    # remove ospf nexthop weight
+                    self.remove_nexthop()
+
+                if not self.area and not self.nexthop_addr:
+                    # remove ospf process
+                    self.delete_process()
+            else:
+                self.module.fail_json(msg='Error: ospf process does not exist')
+
+        self.get_end_state()
+        self.results['changed'] = self.changed
+        self.results['proposed'] = self.proposed
+        self.results['existing'] = self.existing
+        self.results['end_state'] = self.end_state
+        if self.changed:
+            self.results['updates'] = self.updates_cmd
+        else:
+            self.results['updates'] = list()
+
+        self.module.exit_json(**self.results)
+
+
+def main():
+    """Module main"""
+
+    argument_spec = dict(
+        process_id=dict(required=True, type='str'),
+        area=dict(required=False, type='str'),
+        addr=dict(required=False, type='str'),
+        mask=dict(required=False, type='str'),
+        auth_mode=dict(required=False,
+                       choices=['none', 'hmac-sha256', 'md5', 'hmac-md5', 'simple'], type='str'),
+        auth_text_simple=dict(required=False, type='str', no_log=True),
+        auth_key_id=dict(required=False, type='str'),
```

Comment:
For integers please:
use `type='int'`
remove the `isdigit` check from `check_params`